src/object_store/src/object/mod.rs

```diff
@@ -738,19 +730,20 @@ pub async fn parse_remote_object_store(
                     .monitored(metrics),
             )
         }
-        s3_compatible if s3_compatible.starts_with("s3-compatible://") => {
-            ObjectStoreImpl::S3Compatible(
-                S3ObjectStore::new_s3_compatible(
-                    s3_compatible
-                        .strip_prefix("s3-compatible://")
-                        .unwrap()
-                        .to_string(),
-                    metrics.clone(),
-                )
-                .await
-                .monitored(metrics),
+
+        s3_compatible if s3_compatible.starts_with("s3-compatible://") => ObjectStoreImpl::S3(
+            // For backward compatibility, s3-compatible is still reserved.
```

Comment:
Do we want100% compatibility? If so, we also need to set the new envs if the old envs are set.